src/olympia/files/models.py

```diff
@@ -42,9 +42,7 @@ class File(OnChangeMixin, ModelBase):
     id = PositiveAutoField(primary_key=True)
     STATUS_CHOICES = amo.STATUS_CHOICES_FILE
 
-    version = models.ForeignKey(
-        'versions.Version', related_name='files', on_delete=models.CASCADE
-    )
+    version = models.OneToOneField('versions.Version', on_delete=models.CASCADE)
```

Comment:
Something I noticed during fixing the tests was `SaveUpdateMixin.reload()` doesn't handle reverse relations for `OneToOneField`s - i.e. `version.reload` doesn't reload `version.file` - you have to specifically do `version.file.reload()`.  We should look into dropping our special sauce and just use django's `.refresh_from_db`